content/blog/2018/02/2018-02-22-cheetah.adoc

```diff
@@ -0,0 +1,148 @@
+---
+:layout: post
+:title: "Project Cheetah - Faster, Leaner Pipeline That Can Keep Up With Demand"
+:tags:
+- pipeline
+- performance
+- scalability
+:author: svanoort
+---
+
+:toc: 
+
+[NOTE]
+====
+This is a guest post by link:https://github.com/svanoort[Sam Van Oort],
+Software Engineer at link:http://cloudbees.com[CloudBees], and contributor to
+the Jenkins project, and maintainer of the Pipeline Plugins.
+====
+
+Since it launched, Pipeline has had a bit of a Dr. Jekyll and Mr. Hyde performance problem.  In certain circumstances, Pipeline can turn from a mild-mannered CI/CD assistant into a monster.  It will happily eat storage read/write capacity like popcorn without caring about the other concerns of our friendly butler.  When combined with other additional factors, this can result in real-world stability problems.  For example, combining slow storage with a spike in running Pipelines has brought down production Jenkins at more than one organization.  Similarly, users see issues if a busy master gets hit with an extra source of stress; past culprits have been heavy (ab)use of automated APIs, now-solved user lookup bugs, backup jobs, and plugins run amuck that load excessive numbers of builds.  Symptoms ranged from visible slowdowns in the UI to unresponsive jobs and "hung" masters.
```

Comment:
It's not entirely clear to me what's meant by "automated APIs" — you mean people hitting the Jenkins API endpoints?

I'd say "run amok" rather than "run amuck" — I _think_ that's outdated.

Reply:
I did an edit that should make that a bit cleared (see latest)